### PR TITLE
ISPN-9373 Upgrade testng to 6.14.3

### PR DESCRIPTION
**Pull request by diegolovison**
https://issues.jboss.org/browse/ISPN-9373 Bump testng

### CODE REVIEW

**Comment by pruivo** [comment]
needs rebase :)

**Comment by diegolovison** [comment]
Please don't integrate yet. I would like to compare the Jenkins Test Result with the commit d0bef987f5

**Comment by diegolovison** [comment]
I am able to reproduce "some" errors on my local. I will work on that.

**Comment by pruivo** [comment]
why adding `junit` dependency? we use `testng` in most of the places :)

**Comment by diegolovison** [comment]
@pruivo with the new testng version, the "junit dependency" goes away. I would like to keep the code as close as possible to reduce the noise.

For example: `FunctionalMapTest` it is failing only when you run: `mvn verify -Dtest=FunctionalMapTest`. If you try to debug it works. if you run one method only, it works.

I can remove the `junit` dependency when the test start working again.

**Comment by diegolovison** [comment]
Please don't integrate yet.

File: integrationtests/security-it/pom.xml [hunk 1]
```diff
@@ -83,13 +83,7 @@
...
             <artifactId>maven-failsafe-plugin</artifactId>
             <configuration>
-               <!-- Force use of JUnit, since TestNG+Arquillian break in wonderful ways -->
-               <testNGArtifactName>none:none</testNGArtifactName>
-               <forkCount>1</forkCount>
-               <reuseForks>false</reuseForks>
-               <properties>
-                  <listener>${junitListener}</listener>
-               </properties>
+               <forkMode>once</forkMode>
```
Comment (danberindei) [inline integrationtests/security-it/pom.xml]: `forkMode` is [deprecated](http://maven.apache.org/surefire/maven-surefire-plugin/test-mojo.html#forkMode), and are you sure we don't need to force the use of JUnit any more?

File: spring/spring4/spring4-remote/src/test/java/org/infinispan/spring/provider/SpringRemoteCacheManagerFactoryBeanContextTest.java [hunk 2]
```diff
@@ -1,6 +1,6 @@
 package org.infinispan.spring.provider;
 
-import static org.junit.Assert.assertEquals;
+import static org.testng.Assert.assertEquals;
```
Comment (danberindei) [inline spring/spring4/spring4-remote/src/test/java/org/infinispan/spring/provider/SpringRemoteCacheManagerFactoryBeanContextTest.java]: We should use `AssertJUnit`, which is part of TestNG but keeps the same parameters as the JUnit `Assert`

**Comment by danberindei** [comment]
> @pruivo with the new testng version, the "junit dependency" goes away. 

It took me a while to parse this sentence, with surefire selecting the provider based on what's on the classpath I never would have thought that TestNG has a dependency on junit :)

Anyway, we should remove the dependency, but we should be using `org.testng.AssertJUnit` instead of `org.testng.Assert`, because we still have dreams of switching to JUnit everywhere in the future.

**Comment by tristantarrant** [comment]
Can this be updated ?

**Comment by diegolovison** [comment]
There are some changes like
```
org.junit.Assert.assertEquals
```
to
```
org.testng.AssertJUnit.assertEquals
```

Testng had junit as a dependency, now with the version `6.14.3` the Junit dependency didn't exists.
Should I add the Junit dependency in the projects?

**Comment by danberindei** [comment]
I say no, we should use `AssertJUnit`

**Comment by tristantarrant** [comment]
AssertJUnit everywhere please.

**Comment by diegolovison** [comment]
Cool. My PR is ready for review

File: jcache/embedded/src/test/java/org/infinispan/jcache/LimitExpiryFactoryTest.java [hunk 3]
```diff
@@ -1,14 +1,15 @@
-package org.infinispan.jcache.embedded;
+package org.infinispan.jcache;
```
Comment (diegolovison) [inline jcache/embedded/src/test/java/org/infinispan/jcache/LimitExpiryFactoryTest.java]: Weird change

**Comment by diegolovison** [comment]
Ready for review

**Comment by danberindei** [comment]
Not sure if you've noticed @diegolovison @tristantarrant, but the all the test methods are now displayed in IntelliJ as `TestClassName (number)`.

`FunctionalListenersTest` is also failing since the upgrade, probably because the newer TestNG is more strict about running configuration methods from another group.

**Comment by diegolovison** [comment]
@danberindei 

Adding the following methods in `FunctionalListenersTest`, the NPE from `mvn verify -Dtest=FunctionalListenersTest` went away

```java
   @Override
   @BeforeClass
   public void createBeforeClass() throws Throwable {
      super.createBeforeClass();
   }

   @Override
   @BeforeMethod
   public void createBeforeMethod() throws Throwable {
      super.createBeforeMethod();
   }
```

But, it doesn't fix the IntelliJ issue

**Comment by diegolovison** [comment]
About IntelliJ see: https://youtrack.jetbrains.com/issue/IDEA-208083

**Pull request merged**